plugins/cache_promote/CMakeLists.txt

```diff
@@ -0,0 +1,24 @@
+#######################
+#
+#  Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+#  agreements.  See the NOTICE file distributed with this work for additional information regarding
+#  copyright ownership.  The ASF licenses this file to you under the Apache License, Version 2.0
+#  (the "License"); you may not use this file except in compliance with the License.  You may obtain
+#  a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing permissions and limitations under
+#  the License.
+#
+#######################
+
+add_atsplugin(cache_promote
+        cache_promote.cc
+        configs.cc
+        policy.cc
+        lru_policy.cc
+        policy_manager.cc
```

Comment:
Out of curiosity: how do header files relate to this? Will CMake detect changes in header files and rebuild the required files?

Reply:
Yes! I did an experiment where I just changed a random header file.  It recompiled .cc files that included that, recreated libraries and then rebuild other targets that depended on that target (even transitive ones)